libgraph/include/katana/GraphTopology.h

```diff
@@ -172,8 +177,47 @@ class KATANA_EXPORT GraphTopology : public GraphTopologyTypes {
     return edge_property_index(eid);
   }
 
+  // We promote these methods to the base topology to make it easier to search among all
+  // already cached topologies using the same predicate.
+  bool is_transposed() const noexcept {
+    return has_transpose_state(tsuba::RDGTopology::TransposeKind::kYes);
+  }
+
+  bool has_transpose_state(
+      const tsuba::RDGTopology::TransposeKind& expected) const noexcept {
+    return tpose_state_ == expected;
+  }
+
+  tsuba::RDGTopology::TransposeKind transpose_state() const noexcept {
+    return tpose_state_;
+  }
+
+  tsuba::RDGTopology::EdgeSortKind edge_sort_state() const noexcept {
+    return edge_sort_state_;
+  }
+
+  bool has_nodes_sorted_by(
+      const tsuba::RDGTopology::NodeSortKind& kind) const noexcept {
+    if (kind == tsuba::RDGTopology::NodeSortKind::kAny) {
+      return true;
+    }
+    return node_sort_state_ == kind;
+  }
+
+  tsuba::RDGTopology::NodeSortKind node_sort_state() const noexcept {
+    return node_sort_state_;
+  }
+
   void Print() const noexcept;
 
+protected:
+  tsuba::RDGTopology::TransposeKind tpose_state_ =
+      tsuba::RDGTopology::TransposeKind::kNo;
+  tsuba::RDGTopology::EdgeSortKind edge_sort_state_ =
+      tsuba::RDGTopology::EdgeSortKind::kAny;
+  tsuba::RDGTopology::NodeSortKind node_sort_state_ =
+      tsuba::RDGTopology::NodeSortKind::kAny;
+
```

Comment:
Initialize with braces rather than assignment

Nit: it is better to have these private and mutable accessors as protected

Reply:
> Nit: it is better to have these private and mutable accessors as protected
How so? That's effectively the same as what we have here.